lib/teleterm/gateway/gateway_kube.go

```diff
@@ -163,5 +170,12 @@ func (g *Gateway) writeKubeconfig(key *keys.PrivateKey, cas map[string]tls.Certi
 	}
 
 	config := kubeconfig.CreateLocalProxyConfig(clientcmdapi.NewConfig(), values)
-	return trace.Wrap(kubeconfig.Save(g.KubeconfigPath(), *config))
+	if err := kubeconfig.Save(g.KubeconfigPath(), *config); err != nil {
+		return trace.Wrap(err)
+	}
+
+	g.cleanupFuncs = append(g.cleanupFuncs, func() error {
+		return trace.Wrap(utils.RemoveFileIfExist(g.KubeconfigPath()))
```

Comment:
not found error kubectl example:
```
$ KUBECONFIG=not_found kubectl get pod
W0705 15:54:05.352504   28156 loader.go:222] Config not found: not_found
E0705 15:54:05.355350   28156 memcache.go:238] couldn't get current server API group list: Get "http://localhost:8080/api?timeout=32s": dial tcp [::1]:8080: connect: connection refused
E0705 15:54:05.355788   28156 memcache.go:238] couldn't get current server API group list: Get "http://localhost:8080/api?timeout=32s": dial tcp [::1]:8080: connect: connection refused
E0705 15:54:05.356873   28156 memcache.go:238] couldn't get current server API group list: Get "http://localhost:8080/api?timeout=32s": dial tcp [::1]:8080: connect: connection refused
E0705 15:54:05.358003   28156 memcache.go:238] couldn't get current server API group list: Get "http://localhost:8080/api?timeout=32s": dial tcp [::1]:8080: connect: connection refused
E0705 15:54:05.359051   28156 memcache.go:238] couldn't get current server API group list: Get "http://localhost:8080/api?timeout=32s": dial tcp [::1]:8080: connect: connection refused
The connection to the server localhost:8080 was refused - did you specify the right host or port?
```

local proxy down kubectl example:
```
The connection to the server teleport.dev.aws.stevexin.me:443 was refused - did you specify the right host or port?
```

The reason I added the cleanup is to avoid keeping unused files around forever, since I added `ConfigDir`. If it's in tsh home, it will get cleanup eventually.